src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,58 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname) {
+        char *promote_secondaries_sysctl = NULL, *promote_secondaries_path = NULL;
+        int r;
+
+        promote_secondaries_path = strjoina("net/ipv4/conf/", ifname, "/promote_secondaries");
+        r = sysctl_read(promote_secondaries_path, &promote_secondaries_sysctl);
+        if (r < 0) {
+                log_debug_errno(r, "cannot read %s", promote_secondaries_path);
```

Comment:
The message should be capitalized. Also, "Cannot read net/ipv4/conf..." is misleading, because it's not an absolute path. Please either make it "Cannot read sysctl net/ipv4/conf..." or otherwise indicate what the path is.